addons/project/tests/test_project_subtasks.py

```diff
@@ -157,22 +139,22 @@ def test_subtask_display_project(self):
                 subtask_form.parent_id = self.env['project.task']
             orphan_subtask = subtask_form.save()
 
-            self.assertEqual(orphan_subtask.display_project_id, self.project_goats, "Display Project of the task should be well assigned")
-            self.assertEqual(orphan_subtask.project_id, self.project_goats, "Changing display project id on a subtask should change project id")
+            self.assertEqual(orphan_subtask.project_id, self.project_goats, "The project of the orphan task should stay the same even if the subtask has no longer parent task")
             self.assertFalse(orphan_subtask.parent_id, "Parent should be false")
 
             # 7)
             with Form(self.task_1.with_context({'tracking_disable': True})) as task_form:
                 with task_form.child_ids.new() as child_task_form:
                     child_task_form.name = 'Test Subtask 1'
-                    child_task_form.display_project_id = self.project_goats
+                    child_task_form.project_id = self.project_goats
             with Form(self.task_1.child_ids.with_context({'tracking_disable': True})) as subtask_form:
-                subtask_form.display_project_id = self.env['project.project']
-                subtask_form.parent_id = self.env['project.task']
+                subtask_form.project_id = self.env['project.project']
+                with self.assertRaises(AssertionError, msg="the parent_id field should not be visible since no project is set on the subtask and so allow_subtasks is false"):
+                    subtask_form.parent_id = self.env['project.task']
             orphan_subtask = subtask_form.save()
 
-            self.assertEqual(orphan_subtask.project_id, self.project_pigs, "Removing parent should not change project")
-            self.assertEqual(orphan_subtask.display_project_id, self.project_pigs, "Removing parent should make the display project set as project.")
+            self.assertFalse(orphan_subtask.project_id, "The project should be removed as expected.")
+            self.assertEqual(orphan_subtask.parent_id, self.task_1, "The parent should not be removed since the field was not there in the form view after project has been removed.")
```

Comment:
```suggestion
            self.assertEqual(orphan_subtask.parent_id, self.task_1, "The parent should not be removed since the field was not there in the form view after project had been removed.")
```
:angel: